Libraries/MiscDrivers/Display/display.h

```diff
@@ -0,0 +1,65 @@
+/******************************************************************************
+ * Copyright (C) 2022 Maxim Integrated Products, Inc., All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL MAXIM INTEGRATED BE LIABLE FOR ANY CLAIM, DAMAGES
+ * OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Except as contained in this notice, the name of Maxim Integrated
+ * Products, Inc. shall not be used except as stated in the Maxim Integrated
+ * Products, Inc. Branding Policy.
+ *
+ * The mere transfer of this software does not imply any licenses
+ * of trade secrets, proprietary technology, copyrights, patents,
+ * trademarks, maskwork rights, or any other form of intellectual
+ * property whatsoever. Maxim Integrated Products, Inc. retains all
+ * ownership rights.
+ *
+ ******************************************************************************/
+
+#ifndef DRIVER_DISPLAY_H_
+#define DRIVER_DISPLAY_H_
+
+#define DISP_E_SUCCESS 0
+#define DISP_E_ERROR -1
+#define DISP_E_BUSY -99
+#define DISP_E_BAD_STATE -98
+#define DISP_E_TIME_OUT -97
+#define DISP_E_BAD_PARAM -96
+#define DISP_E_NOT_CONFIGURED -95
+
+typedef struct {
+    uint16_t x1;
+    uint16_t y1;
+    uint16_t x2;
+    uint16_t y2;
+} display_area_t;
+
+typedef enum {
+    DISPLAY_NORMAL,
+    DISPLAY_FLIP,
+    DISPLAY_ROTATE,
+} display_rotation_t;
+
+typedef struct {
+    int (*init)(void);
+    int (*write)(uint8_t *data, uint32_t data_len);
+    uint8_t *comm_buffer;
+    uint32_t comm_buffer_len;
+} display_comm_api;
```

Comment:
Re-adding a common display header is a good change.  I think we should consider this for use with the other displays and think about the common API a little more.

I have implemented the following scheme for the st7789v drivers:
* Declare SPI transport layer functions as "extern" (Init/Read/Write):  https://github.com/Analog-Devices-MSDK/msdk/blob/7c709440f43bec33167c26b1491d0017084edee0/Libraries/MiscDrivers/Display/tft_st7789v.h#L148
* Implement them in board.c:  https://github.com/Analog-Devices-MSDK/msdk/blob/7c709440f43bec33167c26b1491d0017084edee0/Libraries/Boards/MAX78000/EvKit_V1/Source/board.c#L92

This is very similar to the approach you have taken here.  So I wonder if we can consolidate the other drivers to also use this common header, and choose between the "extern" approach or this common struct with function pointers.

Reply:
Lets apply this can by another dedicated PR.